src/Mvc/Mvc.Analyzers/src/Startup/MiddlewareAnalyzer.cs

```diff
@@ -0,0 +1,48 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Collections.Immutable;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.Operations;
+
+namespace Microsoft.AspNetCore.Analyzers
+{
+    internal class MiddlewareAnalyzer
+    {
+        private readonly StartupAnalyzerContext _context;
+
+        public MiddlewareAnalyzer(StartupAnalyzerContext context)
+        {
+            _context = context;
+        }
+
+        public void AnalyzeConfigureMethod(OperationBlockStartAnalysisContext context)
+        {
+            var configureMethod = (IMethodSymbol)context.OwningSymbol;
+            var middleware = ImmutableArray.CreateBuilder<MiddlewareItem>();
+
+            // Note: this is a simple source-order implementation. We don't attempt perform data flow
+            // analysis in order to determine the actual order in which middleware are ordered.
+            //
+            // This can currently be confused by things like Map(...)
```

Comment:
THIS IS SUPER HACK MODE 😆

Reply:
There's an existing package from Roslyn that we can use to do more sophicsticated analysis, but it's not needed in this PR.